cloudbridge/cloud/providers/azure/azure_client.py

```diff
@@ -246,9 +248,15 @@ def network_management_client(self):
     @property
     def blob_service(self):
         if not self._block_blob_service:
-            self._block_blob_service = BlockBlobService(
-                self.storage_account,
-                self.access_key_result.keys[0].value)
+            if self._access_token:
+                token_credential = TokenCredential(self._access_token)
+                self._block_blob_service = BlockBlobService(
+                    account_name=self.storage_account,
+                    token_credential=token_credential)
+            else:
+                self._block_blob_service = BlockBlobService(
+                    self.storage_account,
+                    self.access_key_result.keys[0].value)
```

Comment:
it would be great if we could use named parameters here too, so that it's clear what the difference between the two variants.

Reply:
Agree, added.